### DISCUSSION TITLE
remove() Function to Remove Config Params from Existing Config Array

### DISCUSSION THREAD

**Discussion by sprankhub** [deployphp/deployer#3889] [Features]
Wouldn't it make sense to provide `remove()`, similar to [`add()`](https://deployer.org/docs/7.x/api#add)? Usage example:

Core recipe (from the magento2 recipe):

    set('shared_dirs', [
        'var/composer_home',
        'var/log',
        'var/export',
        'var/report',
        'var/import',
        'var/import_history',
        'var/session',
        'var/importexport',
        'var/backups',
        'var/tmp',
        'pub/sitemap',
        'pub/media'
    ]);

Inside my `deploy.php`:

    remove('shared_dirs', 'pub/sitemap');

Resulting config value for `shared_dirs`:

    [
        'var/composer_home',
        'var/log',
        'var/export',
        'var/report',
        'var/import',
        'var/import_history',
        'var/session',
        'var/importexport',
        'var/backups',
        'var/tmp',
        'pub/media'
    ]

Since there is `add`, I think it is natural to also have the opposite. Currently, it needs to be done with `get` and `set`.

## Upvote & Fund

- We're using [Polar.sh](https://polar.sh/deployphp) so you can upvote and help fund this issue.
- We receive the funding once the issue is completed & confirmed by you.
- Thank you in advance for helping prioritize & fund our backlog.

<a href="https://polar.sh/deployphp/deployer/issues/3355">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/deployphp/deployer/issues/3355/pledge.svg?darkmode=1">
  <img alt="Fund with Polar" src="https://polar.sh/api/github/deployphp/deployer/issues/3355/pledge.svg">
</picture>
</a>

**Comment by antonmedv** [comment]
How often remove() will be used. While it is possible to add (and super easy). remove also possible to do with, as you mentioned, set&get. 

Do we really need this function? Will it make API better?

**Reply by sprankhub** [reply]
`add` is also possible with `get` and `set`, but we still have it :) I think it makes the API more complete and more straightforward. It is weird having `add`, but not having `remove` / `delete`.

We remove stuff in a few recipes, at least. I'd say it is not too uncommon, but I might be wrong.

**Reply by antonmedv** [reply]
> add is also possible with get and set, but we still have it 

Good point. Well, I think we can add remove to be add mirror. ಠ_ಠ
